File: app/src/main/java/org/descartae/android/view/fragments/empty/EmptyGPSOfflineFragment.java
package org.descartae.android.view.fragments.empty;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.descartae.android.R;
import org.descartae.android.interfaces.RetryConnectionView;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class EmptyGPSOfflineFragment extends Fragment {

    private RetryConnectionView mListener;

    public EmptyGPSOfflineFragment() {
    }

    public static EmptyGPSOfflineFragment newInstance() {
        EmptyGPSOfflineFragment fragment = new EmptyGPSOfflineFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_gps_offl_empty, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof RetryConnectionView) {
            mListener = (RetryConnectionView) context;
        } else {
            throw new RuntimeException(context.toString() + " must implement RetryConnectionView");
        }
    }

    @OnClick(R.id.action_try)
    public void actionTry() {
        mListener.onRetryConnection();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }
}

File: app/src/main/java/org/descartae/android/view/fragments/empty/EmptyGPSOfflineFragment.kt
package org.descartae.android.view.fragments.empty

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_gps_offl_empty.*
import org.descartae.android.R
import org.descartae.android.interfaces.RetryConnectionView


class EmptyGPSOfflineFragment : Fragment() {

    private var mListener: RetryConnectionView? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_gps_offl_empty, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        action_try.setOnClickListener {mListener?.onRetryConnection() }
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is RetryConnectionView) {
            mListener = context
        } else {
            throw RuntimeException(context!!.toString() + " must implement RetryConnectionView")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    companion object {

        fun newInstance(): EmptyGPSOfflineFragment {
            val fragment = EmptyGPSOfflineFragment()
            val args = Bundle()
            fragment.arguments = args
            return fragment
        }
    }
}
